### PR TITLE
Chore/cls2 1354 update investment delivery partners le ps to id ps

### PR DESCRIPTION
**Pull request by marijnkampf**
### Description of change

Command to change LEPs to IDPs.

Use `--simulate` to get summary output of changes without saving them.
Use `--delete` to delete LEPs that have new IDPs assigned to them.

The command will be run in two phases. First to add new IDPs to the Investment Projects. Once they've been verified as being correct the command will be run with --delete which will remove the LEPs of Investment Projects that have the new IDP assigned to it.

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/dbmaintenance/management/commands/update_investment_delivery_partners.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+from django.core.management import BaseCommand
+
+from datahub.investment.project.models import InvestmentProject
+
+logger = getLogger(__name__)
+
+delivery_partner_mappings = [
+    # Cumbria LEP -> Cumbria
+    {'lep': 'e96192bb-09f1-e511-8ffa-e4115bead28a', 'idp': '4d2d0351-ffaa-4a0d-986a-f13be4ec2198'},
```
Comment (baarkerlounger) [inline datahub/dbmaintenance/management/commands/update_investment_delivery_partners.py:14]: Where do these mappings come from?

**Review by baarkerlounger — APPROVED** [review]

File: datahub/dbmaintenance/management/commands/update_investment_delivery_partners.py (hunk 1 above)
Comment (marijnkampf) [reply datahub/dbmaintenance/management/commands/update_investment_delivery_partners.py:14]: There is a table provided in the ticket https://uktrade.atlassian.net/browse/CLS2-1354. Which I mapped to the Investment Delivery Partner records as on the live site in  https://docs.google.com/spreadsheets/d/1yHelzYDoBIdSRgqc8gL9nnxCOBaEDpO8soOm_T2xiBc/edit?gid=1738328916#gid=1738328916

**Review by oliverjwroberts — COMMENTED** [review]
Couple of minor comments, but overall looks good to me! Do we know how many investment projects this is going to modify?

File: datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        )
+        caplog.set_level('INFO')
+
+        InvestmentProjectFactory(
+            actual_land_date=datetime(2025, 4, 1, tzinfo=timezone.utc),
+            delivery_partners=[lep],
+        )
+
+        call_command('update_investment_delivery_partners', simulate=False, delete=True)
+        message = "{'projects': {'count': 0, 'errors': []}, 'leps': {'investment_project_count': 0, 'to_delete': 0, 'deleted': 0, 'errors': []}, 'idps': {'investment_project_count': 0, 'to_add': 0, 'added': 0, 'errors': []}"
```
[the remark this answers is not in the document]
Comment (marijnkampf) [reply datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py:195]: If the LEP doesn't exist there won't be any Investment projects that have it set as a Delivery Partner so no check required. The IDP check *is* required as there is an edge case where an LEP should be replaced by an IDP that isn't there.

File: datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        call_command('update_investment_delivery_partners', simulate=simulate, delete=delete)
+
+        assert caplog_text in caplog.text
+
+    @pytest.mark.django_db
+    def test_lep_does_not_exist(
+        self,
+        mocker,
+        caplog: pytest.LogCaptureFixture,
+        lep: InvestmentDeliveryPartner,
```
[the remark this answers is not in the document]
Comment (marijnkampf) [reply datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py:175]: Ruff added the type so I presumed it was best practice. The `lep` fixture returns an InvestmentDeliveryPartner.

File: datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        message = "{'projects': {'count': 0, 'errors': []}, 'leps': {'investment_project_count': 0, 'to_delete': 0, 'deleted': 0, 'errors': []}, 'idps': {'investment_project_count': 0, 'to_add': 0, 'added': 0, 'errors': []}"
+
+        assert message in caplog.text
+
+    @pytest.mark.django_db
+    def test_idp_does_not_exist(
+        self,
+        mocker,
+        caplog: pytest.LogCaptureFixture,
+        lep: InvestmentDeliveryPartner,
```
[the remark this answers is not in the document]
Comment (marijnkampf) [reply datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py:204]: As above

File: datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    ):
+        """Test idp doesn't exist."""
+        mocker.patch(
+            'datahub.dbmaintenance.management.commands.update_investment_delivery_partners.delivery_partner_mappings',
+            new=[
+                {
+                    'lep': 'abcdef01-09f1-e511-8ffa-e4115bead28a',
```
[the remark this answers is not in the document]
Comment (marijnkampf) [reply datahub/dbmaintenance/test/commands/test_update_investment_delivery_partners.py]: It needs to be a valid uuid. I've added a comment.

**Review by oliverjwroberts — APPROVED** [review]
